draft-irtf-cfrg-opaque.md

```diff
@@ -384,139 +383,223 @@ variant) of the OPRF described in {{I-D.irtf-cfrg-voprf}}.
     This function also needs to satisfy collision resistance.
 
 OPAQUE additionally depends on an Authenticated Key Exchange (AKE) protocol.
-This specification defines one particular AKE based on 3DH; see {{protocol}}.
+This specification defines one particular AKE based on 3DH; see {{ake-protocol}}.
 We let `Npk` and `Nsk` denote the size of public and private keys, respectively,
-used in the AKE.
+used in the AKE. The AKE protocol must provide the following functions:
 
-- Deterministic AKE Key Generation Functions:
-  - RecoverPublicKey(private_key): Recover the public key related to the input private key.
-  - DeriveAkeKeyPair(seed): Derive a private and public key pair deterministically from the seed.
-  - GenerateKeyPair(): Return a randomly generated private and public key pair. This can be
-    implemented by generating a random private key `sk`, then computing `pk = RecoverPublicKey(sk)`.
+- RecoverPublicKey(private_key): Recover the public key related to the input `private_key`.
+- DeriveAuthKeyPair(seed): Derive a private and public authentication key key pair
+  deterministically from the input `seed`.
+- GenerateKeyPair(): Return a randomly generated private and public key pair. This can be
+  implemented by generating a random private key `sk`, then computing `pk = RecoverPublicKey(sk)`.
 
-Random nonces used in this protocol are of length `Nn = 32` bytes.
+Finally, all random nonces used in this protocol are of length `Nn` = 32 bytes.
 
-# Client Credential Storage {#client-credential-storage}
+# Protocol Overview {#protocol-overview}
 
-OPAQUE makes use of a structure `Envelope` to manage client credentials. This envelope holds information
-about its format and content for the client to obtain its authentication material.
+OPAQUE consists of two stages: registration and authenticated key exchange.
+In the first stage, a client registers its password with the server and stores
+its encrypted credentials on the server. The client inputs its credentials,
+which includes its password and user identifier, and the server inputs its
```

Comment:
The phrasing may exclude the external mode. Is it intentional?

Reply:
No, but I didn't want to confuse reads so early on. I figured omitting that for now and explaining internal/external mode later was easiest.